pkg/internal/device/power_meter.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package devices
+
+import "fmt"
+
+// Joules represents an energy value in SI unit of Joule.
+// The value can additionally have an SI prefix of Micro, Milli etc
+// not captured in this type.
+// ** Do not assume that the value contained is a multiple of 1 Joule **,
+// it could be MicroJoules, or MilliJoules according to the interface
+// or method which returns it.
+type Joules uint64
```

Comment:
How about defining Energy then and store `uj` which can then have `Joules`, `MicroJoules` which return different units just like `time.Duration` -> https://pkg.go.dev/time#Duration.Microseconds

Reply:
Fundamental point it that SI unit of Energy is Joule. If want to state that the type is `Energy` (just like `Duration`), then we can define `Energy` Type which has `Joules() -> float64`, `MicroJoules() -> uint64`

Reply:
ok

Reply:
should `Energy` be in its own package? `internal/energy` ?